### PR TITLE
Fixes for top-of-tree LLVM (#6546)

### PR DESCRIPTION
**Pull request by steven-johnson**

### CODE REVIEW

**Comment by steven-johnson** [comment]
Attn @LebedevRI

**Comment by alexreinking** [comment]
Do we need `regexp_replace`, or is `sed` sufficient for the Makefile?

It probably doesn't need to be built by CMake since it's never used.

File: Makefile [hunk 1]
```diff
@@ -1088,6 +1104,10 @@ $(BIN_DIR)/binary2cpp: $(ROOT_DIR)/tools/binary2cpp.cpp
 	@mkdir -p $(@D)
 	$(CXX) $< -o $@
 
+$(BIN_DIR)/regexp_replace: $(ROOT_DIR)/tools/regexp_replace.cpp
+	@mkdir -p $(@D)
+	$(CXX) -std=c++11 $< -o $@
```
Comment (alexreinking) [inline Makefile:1109]: C++17, surely

**Comment by LebedevRI** [comment]
> Attn @LebedevRI

Thank you for looking into it!

**Review by alexreinking — APPROVED** [review]
LGTM modulo language version

**Comment by steven-johnson** [comment]
HVX still broken, needs further investigation

**Pull request merged**